packages/bitcore-wallet-client/src/lib/credentials.ts

```diff
@@ -162,24 +163,10 @@ export class Credentials {
           purpose = '48';
           break;
       }
-
-      var coin = '0';
-      if (this.network != 'livenet' && this.coin !== 'eth') {
-        coin = '1';
-      } else if (this.coin == 'bch') {
-        if (this.use145forBCH) {
-          coin = '145';
-        } else {
-          coin = '0';
-        }
-      } else if (this.coin == 'btc') {
-        coin = '0';
-      } else if (this.coin == 'eth') {
-        coin = '60';
-      } else {
-        throw new Error('unknown coin: ' + this.coin);
+      if (!this.use145forBCH && this.coin === 'bch') {
+        coin = 'btc';
       }
-      return 'm/' + purpose + "'/" + coin + "'/" + this.account + "'";
+      return Deriver.pathFor(coin, this.network, this.account).replace(/44/i, purpose);
```

Comment:
if account is 44, this will be an issue.

Reply:
when coin= BCH,  some wallets uses the `coin` path 0 and other 145. this doesnt seems to take that in account.